mbq/client/contrib/permissions.py

```diff
@@ -132,6 +132,22 @@ def fetch_all_permissions(self, person_id: str) -> FetchedPermissionsDoc:
             raise ServerError("Server error") from e
 
 
+class Registrar:
+    def __init__(self):
+        self._registry: Dict[str, List[Callable[..., Any]]] = {}
+
+    def register(self, name: str, func: Callable[..., Any]) -> None:
+        if name in self._registry:
+            self._registry.get(name).append(func)
+        else:
+            self._registry[name] = [func]
+
+    def emit(self, name: str, *args, **kwargs) -> None:
+        if name in self._registry:
+            for function in self._registry.get(name):
```

Comment:
You won't need to do this once you have THE POWER OF DEFAULTDICT, but the first two lines are more succinctly:

```
for fn in self._register.get(name, []):
```